libkbfs/folder_branch_ops.go

```diff
@@ -805,6 +805,13 @@ func (fbo *folderBranchOps) setHeadLocked(
 			fbo.updateDoneChan = make(chan struct{})
 			go fbo.registerAndWaitForUpdates()
 		}
+		// If journaling is enabled, we should make sure to enable it
+		// for this TLF.  That's because we may have received the TLF
+		// ID from the service, rather than via a GetIDForHandle call,
+		// and so we might have skipped the journal.
+		if jServer, err := GetJournalServer(fbo.config); err == nil {
```

Comment:
Might want to add in a check that the user is a writer.

Reply:
`getTlfJournal` already checks that.

Reply:
Right, thanks.